### ISSUE TITLE
Resource group is not available when creating a new VM

### ISSUE THREAD

**Issue by aovechki** [Microsoft/azure-tools-for-java#298]
win10, x64, Eclipse.Mars

When trying to create a new VM a resource group that was previously created in the same session using the Server Explorer (via create new storage account) is not available for selection. Only pre-existing resource groups are available. Looks to be a caching issue.

**Comment by martinsawicki** [comment]
sounds famliar...

**Comment by lenala** [comment]
Fixed, please verify.

**Comment by davidliangze** [comment]
Build used in Eclipse:WindowsAzurePlugin4EJ.307.04-16-2017
Build used in IntelliJ: azure-toolkit-for-intellij.2017.303.04-16-2017

This issue #233 is fixed, but this issue still repros in Eclipse Neon and Eclipse Mars (in IntelliJ it works fine).

The resource group created during the process of creating a new VM( creating a new resource group for VM or creating a new storage account then create a new resource group for this storage account ) is not available in the dropdown list in Create new Virtual Machine dialog or in Create new Storage Account dialog, unless refreshing the top node in the Azure Explorer.

**Comment by davidliangze** [comment]
OS: Windows 10 Enterprise x64 ENU
IDE: Eclipse Neon/ Mars
Plugin Version: WindowsAzurePlugin4EJ.319.04-17-2017

Test in Eclipse Neon passed, but the issue above repros in Eclipse Mars.

**Comment by lenala** [comment]
Cache for resource group gets updated only after vm/storage account successfully created, might it be the case that creation process still running?

**Comment by davidliangze** [comment]
I've recheck it after a few hours without the top node refreshed. The created resource group is still not available in the dropdown list, but obviously the VM creation is already done.

**Comment by lenala** [comment]
Cannot repro, works for me with Eclipse Mars and latest build.

**Comment by davidliangze** [comment]
Test passed with the latest build.

OS: Windows 10 Enterprise x64 ENU
IDE: Eclipse Mars.2
Plugin Version: WindowsAzurePlugin4EJ.321.04-17-2017
